### PR TITLE
Story/cite 41

### PR DESCRIPTION
**Pull request by MaulikLim**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 

[CITE-41] - It should be possible to filter citations by concepts.
  
## Anything else the reviewer needs to know?

previous branch - https://github.com/diging/citesphere/tree/story/CITE-41

[CITE-41]: https://diging.atlassian.net/browse/CITE-41?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: citesphere/pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
         <spring.kafka.version>2.2.6.RELEASE</spring.kafka.version>
         <spring-social-zotero.version>0.10</spring-social-zotero.version>
         <citesphere.messages.version>0.4</citesphere.messages.version>
-        <citesphere.model.version>1.12</citesphere.model.version>
+        <citesphere.model.version>1.13</citesphere.model.version>
```
Comment (jdamerow) [inline citesphere/pom.xml:28]: I believe this needs to be 1.14?

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: citesphere/pom.xml (hunk 1 above)
Comment (MaulikLim) [reply citesphere/pom.xml]: This has been fixed now

**Comment by jdamerow** [comment]
Resolve conflicts please

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
There are compilation errors:
```
[ERROR] /var/lib/jenkins/workspace/citesphere_deploy_to_review_on_pr/citesphere/src/test/java/edu/asu/diging/citesphere/core/export/impl/ExportManagerTest.java:[88,37] method getGroupItems in interface edu.asu.diging.citesphere.core.service.ICitationManager cannot be applied to given types;
  required: edu.asu.diging.citesphere.user.IUser,java.lang.String,java.lang.String,int,java.lang.String,java.util.List<java.lang.String>
  found: edu.asu.diging.citesphere.user.IUser,java.lang.String,<nulltype>,int,java.lang.String
  reason: actual and formal argument lists differ in length
```

**Comment by MaulikLim** [comment]
Fixed test cases

**Comment by jdamerow** [comment]
- let's make the concepts checkboxes, so it's obvious they can be checked/unchecked
- there should be a select/deselect all button
- there should be a filter to search within the concept
- not sure what currently happens if the list gets very long, but there should be a scrollbar (it shouldn't just keep going).
- the dropdown menu should say "Filter by Concept"

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Pull request merged**